src/actions/mover.ts

```diff
@@ -0,0 +1,334 @@
+      keyCodes: [
+        createSerializedKey(KeyCodes.DOWN, [KeyCodes.ALT]),
+        createSerializedKey(KeyCodes.DOWN, [KeyCodes.CTRL]),
+      ],
+    },
+  ];
+
+  menuItems: ContextMenuRegistry.RegistryItem[] = [
+    {
+      displayText: 'Move Block (M)',
+      preconditionFn: (scope) => {
+        const workspace = scope.block?.workspace as WorkspaceSvg | null;
+        if (!workspace) return 'hidden';
+        return this.canMove(workspace) ? 'enabled' : 'disabled';
+      },
+      callback: (scope) => {
+        const workspace = scope.block?.workspace as WorkspaceSvg | null;
+        if (!workspace) return false;
+        this.startMove(workspace);
+      },
+      scopeType: ContextMenuRegistry.ScopeType.BLOCK,
+      id: 'move',
+      weight: 8.5,
+    },
+  ];
+
+  /**
+   * Install the actions as both keyboard shortcuts and (where
+   * applicable) context menu items.
+   */
+  install() {
+    for (const shortcut of this.shortcuts) {
+      ShortcutRegistry.registry.register(shortcut);
+    }
+    for (const menuItem of this.menuItems) {
+      ContextMenuRegistry.registry.register(menuItem);
+    }
+  }
+
+  /**
+   * Uninstall these actions.
+   */
+  uninstall() {
+    for (const shortcut of this.shortcuts) {
+      ShortcutRegistry.registry.unregister(shortcut.name);
+    }
+    for (const menuItem of this.menuItems) {
+      ContextMenuRegistry.registry.unregister(menuItem.id);
+    }
+  }
+
+  /**
+   * Returns true iff we are able to begin moving the block which
+   * currently has focus on the given workspace.
+   *
+   * @param workspace The workspace to move on.
+   * @returns True iff we can begin a move.
+   */
+  canMove(workspace: WorkspaceSvg) {
+    const block = this.getCurrentBlock(workspace);
+
+    return !!(
+      this.navigation.getState(workspace) === Constants.STATE.WORKSPACE &&
+      this.canEdit(workspace) &&
+      !this.moves.has(workspace) && // No move in progress.
+      block?.isMovable()
+    );
+  }
+
+  /**
+   * Returns true iff we are currently moving a block on the given
+   * workspace.
+   *
+   * @param workspace The workspace we might be moving on.
+   * @returns True iff we are moving.
+   */
+  isMoving(workspace: WorkspaceSvg) {
+    return this.canEdit(workspace) && this.moves.has(workspace);
+  }
+
+  /**
+   * Start moving the currently-focused item on workspace, if
+   * possible.
+   *
+   * Should only be called if canMove has returned true.
+   *
+   * @param workspace The workspace we might be moving on.
+   * @returns True iff a move has successfully begun.
+   */
+  startMove(workspace: WorkspaceSvg) {
+    const cursor = workspace?.getCursor();
+    const block = this.getCurrentBlock(workspace);
+    if (!cursor || !block) throw new Error('precondition failure');
+
+    // Select and focus block.
+    common.setSelected(block);
+    cursor.setCurNode(ASTNode.createBlockNode(block));
+
+    // Additional implementation goes here.
+    console.log('startMove');
```

Comment:
Remove? Applies throughout these next few methods.

Reply:
I filed https://github.com/google/blockly-keyboard-experimentation/issues/357 to track removing all of these if we don't need them. I  think the point was to support multiple types of Movers in case we want to compare two different types of behaviour, but I'm not sure if we'll need to do that.

@cpcallen to confirm.

The same applies for `finishMove` and `abortMove`, which ar ethe same here but distinct in the drag mover.